File: lms-BE/app/features/auth/schemas.py
from datetime import datetime
from typing import List
from pydantic import BaseModel, EmailStr, Field

class LoginRequest(BaseModel):
    email: str
    password: str

class TokenResponse(BaseModel):
    access_token: str
    refresh_token: str
    token_type: str = "bearer"

class RefreshRequest(BaseModel):
    refresh_token: str

class LogoutRequest(BaseModel):
    refresh_token: str

class ChangePasswordRequest(BaseModel):
    email: EmailStr | None = None
    current_password: str = Field(..., min_length=6)
    new_password: str = Field(..., min_length=8)


class UserInfo(BaseModel):
    id: int
    name: str
    email: EmailStr
    role: str

class PasswordChangeRequestRead(BaseModel):
    id: int
    user_id: int
    status: str
    created_at: datetime
    resolved_at: datetime | None = None
    user: UserInfo

    class Config:
        from_attributes = True

class PaginatedPasswordChangeRequests(BaseModel):
    items: List[PasswordChangeRequestRead]
    total: int
    page: int
    size: int

File: lms-FE/src/features/auth/schemas.ts
import { z } from 'zod';

export const loginSchema = z.object({
    email: z.string().email('Please enter a valid email address'),
    password: z.string().min(6, 'Password must be at least 6 characters'),
});

export type LoginFormData = z.infer<typeof loginSchema>;

export interface TokenResponse {
    access_token: string;
    refresh_token: string;
    token_type: string;
}

export const changePasswordSchema = z.object({
    email: z.string().email('Please enter a valid email address').optional(),
    current_password: z.string().min(6, 'Password must be at least 6 characters'),
    new_password: z.string().min(8, 'Password must be at least 8 characters'),
    confirm_password: z.string()
}).refine((data) => data.new_password === data.confirm_password, {

    message: "New passwords don't match",
    path: ["confirm_password"],
});

export type ChangePasswordData = z.infer<typeof changePasswordSchema>;

export interface UserInfo {
    id: number;
    name: string;
    email: string;
    role: string;
}

export interface PasswordChangeRequestRead {
    id: number;
    user_id: number;
    status: 'pending' | 'approved' | 'rejected';
    created_at: string;
    resolved_at: string | null;
    user: UserInfo;
}

export interface PaginatedPasswordChangeRequests {
    items: PasswordChangeRequestRead[];
    total: number;
    page: number;
    size: number;
}
